### ISSUE TITLE
Osdk with Intel TDX config not working.

### ISSUE THREAD

**Issue by Hsy-Intel** [asterinas/asterinas#686]
The Intel TDX qemu configs are somewhat different from the default qemu configs, the qemu running command line:

```shell
/usr/libexec/qemu-kvm 
-accel kvm 
-name process=tdxvm,debug-threads=on 
-m 8G 
-vga none 
-monitor pty 
-no-hpet 
-nodefaults 
-drive file=/home/data/asterinas.qcow2,if=virtio,format=qcow2 
-monitor telnet:127.0.0.1:9003,server,nowait 
-bios /home/siyuan/tdx-tools/OVMF_VARS.fd 
-object tdx-guest,sept-ve-disable,id=tdx,quote-generation-service=vsock:2:4050 
-cpu host,-kvm-steal-time,pmu=off,tsc-freq=1000000000 
-machine q35,kernel_irqchip=split,confidential-guest-support=tdx 
-device virtio-net-pci,netdev=mynet0,disable-legacy=on,disable-modern=off 
-device virtio-keyboard-pci,disable-legacy=on,disable-modern=off 
-device virtio-blk-pci,bus=pcie.0,addr=0x6,drive=x0,disable-legacy=on,disable-modern=off 
-drive file=fs.img,if=none,format=raw,id=x0 
-smp 1 
-netdev user,id=mynet0,hostfwd=tcp::10027-:22,hostfwd=tcp::54136-:8090, 
-chardev stdio,id=mux,mux=on,logfile=/home/siyuan/tdx-tools/vm_log_2024-03-18T0924.log 
-device virtio-serial,romfile= 
-device virtconsole,chardev=mux 
-monitor chardev:mux 
-serial chardev:mux 
-nographic
```

When I add these configs to Osdk.toml, I get an error:
```shell
[root@OCSBESRHLREPO01 my-first-os]# cargo osdk run
[Error]: `-cpu host,-kvm-steal-time,pmu=off,tsc-freq=1000000000` is not allowed to set
```
After deleting the `-cpu` config:

```shell
[root@OCSBESRHLREPO01 my-first-os]# cargo osdk run
[Error]: `-machine q35,kernel_irqchip=split,confidential-guest-support=tdx` is not allowed to set
```

After deleting the `-machine` config:
```shell
[root@OCSBESRHLREPO01 my-first-os]# cargo osdk run
RunConfig { manifest: OsdkManifest { kcmd_args: ["init=/bin/busybox", "path=/usr/local/bin", "--", "sh", "-l"], initramfs: Some("./build/initramfs.cpio.gz"), boot: Boot { loader: Grub, protocol: LinuxEfiHandover64, grub_mkrescue: Some("/usr/bin/grub-mkrescue"), ovmf: None }, qemu: Qemu { args: ["--no-reboot", "-accel kvm", "-name process=tdxvm,debug-threads=on", "-vga none", "-no-hpet", "-nodefaults", "-m 8G", "-smp 1", "-nographic", "-serial chardev:mux", "-display none", "-monitor pty", "-monitor chardev:mux", "-chardev stdio,id=mux,mux=on,signal=off,logfile=qemu.log", "-device isa-debug-exit,iobase=0xf4,iosize=0x04", "-device virtio-net-pci,netdev=mynet0,disable-legacy=on,disable-modern=off", "-device virtio-keyboard-pci,disable-legacy=on,disable-modern=off", "-device virtio-blk-pci,bus=pcie.0,addr=0x6,drive=x0,disable-legacy=on,disable-modern=off", "-device virtio-serial,romfile=", "-device virtconsole,chardev=mux", "-object tdx-guest,sept-ve-disable,id=tdx,quote-generation-service=vsock:2:4050", "-netdev user,id=net01,hostfwd=tcp::36788-:22,hostfwd=tcp::55834-:8080"], machine: Q35, path: Some("/usr/libexec/qemu-kvm") } }, cargo_args: CargoArgs { release: false, features: [] } }
[root@OCSBESRHLREPO01 my-first-os]# 
```
Qemu doesn't seem to be running.

My `Osdk.toml` file:

```toml
kcmd_args = ["init=/bin/busybox", "path=/usr/local/bin"]
init_args = ["sh", "-l"]
initramfs="./build/initramfs.cpio.gz"

[boot]
loader = "grub"
protocol = "linux-efi-handover64"
grub-mkrescue = "/usr/bin/grub-mkrescue"
# ovmf = "/usr/bin/ovmf"

[qemu]
path = "/usr/libexec/qemu-kvm"
machine = "q35"
args = [
    "-accel kvm",
    "-name process=tdxvm,debug-threads=on",
    "-m 8G",
    "-vga none",
    "-monitor pty",
    "-no-hpet",
    "-nodefaults",
    "-drive file=/home/data/asterinas.qcow2,if=virtio,format=qcow2",
    "-monitor telnet:127.0.0.1:9003,server,nowait",
    "-bios /home/siyuan/tdx-tools/OVMF_VARS.fd",
    "-object tdx-guest,sept-ve-disable,id=tdx,quote-generation-service=vsock:2:4050",
    "-device virtio-net-pci,netdev=mynet0,disable-legacy=on,disable-modern=off",
    "-device virtio-keyboard-pci,disable-legacy=on,disable-modern=off",
    "-device virtio-blk-pci,bus=pcie.0,addr=0x6,drive=x0,disable-legacy=on,disable-modern=off",
    "-smp 1",
    "-netdev user,id=net01,hostfwd=tcp::36788-:22,hostfwd=tcp::55834-:8080",
    "-chardev stdio,id=mux,mux=on,signal=off,logfile=qemu.log",
    "-device virtio-serial,romfile=",
    "-device virtconsole,chardev=mux",
    "-monitor chardev:mux",
    "-serial chardev:mux",
    "-nographic",
]
```

**Comment by junyang-zh** [comment]
We treat the CPU and machine arguments because they require specific configurations. The support of TDX specific CPU appended arguments should be hardcoded in the OSDK source. I and @StevenJiang1110 may offer more assistance.

**Comment by junyang-zh** [comment]
Fixed by #704.
